### PR TITLE
Single item page

### PR DESCRIPTION
**Pull request by VanishMax**

### CODE REVIEW

File: src/containers/item/item-content.svelte [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    chosenSize = null;
+    dispatch('change', e.detail);
+  };
+
+  let chosenSize = null;
+  let err = false;
+  let animation = false;
+  const chooseSize = e => chosenSize = e.detail;
+  const purchase = () => {
+    if (!inSizes || (inSizes && chosenSize)) {
```
Comment (aabounegm) [inline src/containers/item/item-content.svelte]:
```suggestion
    if (!inSizes || chosenSize) {
```
looks like someone doesn't remember ITP1, when we had the "implies" operator 😜

File: src/utils/optimal-color.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  color /= 255;
+  if (color <= 0.03928) {
+    return color / 12.92;
+  } else {
+    return Math.pow((color + 0.055) / 1.055, 2.4);
+  }
+};
+
+const luminance = ([r, g, b]) => {
+  return transform(r) * 0.2126 + transform(g) * 0.7152 + transform(b) * 0.0722;
```
[the remark this answers is not in the document]
Comment (VanishMax) [reply src/utils/optimal-color.js]: Nah, it is Lev's magic code and magic is a great helper. May be i'd like to put constants, but i don't know what they really mean :)

File: src/containers/ui/buttons.svelte [hunk 3]
```diff
@@ -20,19 +20,19 @@
...
-      <svg src="images/ui-kit-page/bell.svg" />
+      <svg src="/images/icons/bell.svg" />
     </Button>
     <Button isRound isNormal>
-      <svg src="images/ui-kit-page/x.svg" />
+      <svg src="/images/icons/x.svg" />
     </Button>
-    <Button isOutline isRound isNormal>
-      <svg src="images/ui-kit-page/x.svg" />
+    <Button isOutline isRound normal>
```
[the remark this answers is not in the document]
Comment (VanishMax) [reply src/containers/ui/buttons.svelte]: Lol, happens

File: src/components/ui/chip.svelte [hunk 4]
```diff
@@ -10,6 +10,7 @@
   export let label = '';
   export let name = '';
   export let disabled = false;
+  export let unclickable = false;
   export let multiple = false;
```
[the remark this answers is not in the document]
Comment (VanishMax) [reply src/components/ui/chip.svelte]: The difference between prop naming is written in ui-conventions. Outline: props that affect classnames use 'is' prefix, while logic-oriented props don't have it (unclickable, disabled)

File: src/utils/optimal-color.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (illright) [reply src/utils/optimal-color.js]: I'd love to tell you all what do they mean, but I do not know myself, and I'm pretty sure that there's no meaningful name for them anyway, so just let them be

File: src/containers/item/image-previews.svelte [hunk 5]
```diff
@@ -0,0 +1,96 @@
+<script>
+  import {onMount} from 'svelte';
+  import 'simplebar/dist/simplebar.css';
+  import SimpleBar from 'simplebar';
```
[the remark this answers is not in the document]
Comment (illright) [reply src/containers/item/image-previews.svelte]: The link you sent on CSS tricks suggests to use this very library :)
No, unscrew Edge, it is just as important. And the styling of scrollbars is a terrible mess, so I'm up for a change only if there's a better JS library. I don't trust CSS with this stuff

File: src/components/item/purchase-modal.svelte [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+  const dispatch = createEventDispatcher();
+  const confirm = () => {
+    if (finalPrice >= 0) {
+      closeModal();
+      openSnackbar();
+      dispatch('success');
+    }
+  };
+</script>
+
```
[the remark this answers is not in the document]
Comment (VanishMax) [reply src/components/item/purchase-modal.svelte]: Questions to markup))

File: src/containers/item/item-content.svelte (hunk 1 above)
Comment (VanishMax) [reply src/containers/item/item-content.svelte]: Omg, changes much of my future code))

File: src/components/item/purchase-modal.svelte (hunk 6 above)
[the remark this answers is not in the document]
Comment (illright) [reply src/components/item/purchase-modal.svelte]: More like questions to whoever moved the code from markup to frontend

File: src/utils/optimal-color.js [hunk 7]
```diff
@@ -0,0 +1,76 @@
+const trimHEX = (color) => {
+  if (!color) return '';
+
+  let unhashed = color.replace(/[^0-9A-Fa-f]/g, '');
```
[the remark this answers is not in the document]
Comment (VanishMax) [reply src/utils/optimal-color.js]: We should be ready for any data without throwing errors, so if someone pass a string meaning 'this is a color, man, believe me', we should believe and do everything to find a color in it

File: src/routes/item/[id].svelte [hunk 8]
```diff
@@ -0,0 +1,76 @@
+<script context="module">
+  import items from '@/constants/store/store';
+  export function preload({params}) {
```
[the remark this answers is not in the document]
Comment (VanishMax) [reply src/routes/item/[id].svelte]: Prefetch is totally different. Here you navigate to the route (by server), get parameter and use it. Have no idea how is it possible to implement using prefetch that just tells that you shoul expect some route to be rendered before accessed

File: src/routes/item/[id].svelte [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+  <link rel="preload" href="css/projects/main.css" as="style" />
+
+  <!-- Home styles -->
+  <link rel="preload" href="css/home/main.css" as="style" />
+  <link rel="preload" href="css/home/header.css" as="style" />
+  <link rel="preload" href="css/home/tagline.css" as="style" />
+  <link rel="preload" href="css/home/how-to.css" as="style" />
+  <link rel="preload" href="css/home/options.css" as="style" />
+  <link rel="preload" href="css/home/store.css" as="style" />
+  <link rel="preload" href="css/home/contacts.css" as="style" />
```
[the remark this answers is not in the document]
Comment (VanishMax) [reply src/routes/item/[id].svelte]: Good point for the future investigation (creating a task)

File: src/components/item/purchase-modal.svelte [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+    <div class="identifier">
+      <div class="name">{name}</div>
+      <div class="type">{type}</div>
+    </div>
+
+    <div class="quantity">
+      <Button isRound on:click={dec}>
+        <svg src="/images/icons/minus.svg" />
+      </Button>
+      <TextField value={amount} on:input={(e) => checkAmount(parseInt(e.detail))} isNoSpinner type="number" />
```
[the remark this answers is not in the document]
Comment (VanishMax) [reply src/components/item/purchase-modal.svelte]: To backlog, thanks for noticing

File: src/utils/optimal-color.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (aabounegm) [reply src/utils/optimal-color.js]: What if it is valid CSS color (like "black"), but then this line will turn it to "bac", which will then be interpreted as 3 letter hex? Maybe if it doesn't match the regex for hex color it is best to just return it as it is?

**Pull request merged**